datascience/utils.py

```diff
@@ -41,3 +43,31 @@ def get_production(countries, start_date, end_date, delta):
                               'production': o['production'].values()})
             df = df.append(p)
     return df
+
+# get_exchange
+
+def fetch_exchange(country_code, t):
+    endpoint = 'http://electricitymap.tmrow.co'
+    url = '%s/v1/state' % endpoint
+    params = {
+        'datetime': t.to('utc').isoformat()
+    }
+    obj = r.get(url, params=params).json()
+    if not obj['data']['countries'][country_code]: return
+    return obj['data']['countries'][country_code]
+
+def get_exchange(countries, start_date, end_date):
+    delta = 1440 # Only return exchange for an entire day
+    df = pd.DataFrame(columns=['country','timestamp','country_exchange','value'])
+    time_span = date_range(start_date, end_date,delta)
+    for country in countries:
+        print 'Fetching country %s..' % country
+        for t in time_span:
+            print 'Fetching time %s..' % t
+            o = fetch_exchange(country, t)
+            p = pd.DataFrame({'country': country,
+                              'timestamp': t ,
+                              'country_exchange': o['exchange'].keys(),
+                              'value': o['exchange'].values()})
```

Comment:
Are you sure keys and values are iterated in the same order? I think you have no garantee of the iterator order when iterating over a dict..